--- cdrkit-1.1.11.old/genisoimage/tree.c	2008-10-21 19:57:47.000000000 -0400
+++ cdrkit-1.1.11/genisoimage/tree.c	2013-12-06 00:23:18.489622668 -0500
@@ -1139,8 +1139,9 @@
 scan_directory_tree(struct directory *this_dir, char *path, 
 						  struct directory_entry *de)
 {
-	dir		*current_dir;
+        int             current_file;
 	char		whole_path[path_max];
+        struct dirent  **d_list;
 	struct dirent	*d_entry;
 	struct directory *parent;
 	int		dflag;
@@ -1164,7 +1165,8 @@
 	this_dir->dir_flags |= dir_was_scanned;
 
 	errno = 0;	/* paranoia */
-	current_dir = opendir(path);
+	//current_dir = opendir(path);
+        current_file = scandir(path, &d_list, null, alphasort);
 	d_entry = null;
 
 	/*
@@ -1173,12 +1175,12 @@
 	 */
 	old_path = path;
 
-	if (current_dir) {
+	if (current_file >= 0) {
 		errno = 0;
-		d_entry = readdir(current_dir);
+		d_entry = d_list[0];
 	}
 
-	if (!current_dir || !d_entry) {
+	if (current_file < 0 || !d_entry) {
 		int	ret = 1;
 
 #ifdef	use_libschily
@@ -1191,8 +1193,8 @@
 			de->isorec.flags[0] &= ~iso_directory;
 			ret = 0;
 		}
-		if (current_dir)
-			closedir(current_dir);
+		if(d_list)
+			free(d_list);
 		return (ret);
 	}
 #ifdef	abort_deep_iso_only
@@ -1208,7 +1210,7 @@
 			errmsgno(ex_bad, "use rock ridge extensions via -r or -r,\n");
 			errmsgno(ex_bad, "or allow deep iso9660 directory nesting via -d.\n");
 		}
-		closedir(current_dir);
+		free(d_list);
 		return (1);
 	}
 #endif
@@ -1250,13 +1252,13 @@
 		 * the first time through, skip this, since we already asked
 		 * for the first entry when we opened the directory.
 		 */
-		if (dflag)
-			d_entry = readdir(current_dir);
+		if (dflag && current_file >= 0)
+			d_entry = d_list[current_file];
 		dflag++;
 
-		if (!d_entry)
+		if (current_file < 0)
 			break;
-
+                current_file--;
 		/* ok, got a valid entry */
 
 		/* if we do not want all files, then pitch the backups. */
@@ -1348,7 +1350,7 @@
 		insert_file_entry(this_dir, whole_path, d_entry->d_name);
 #endif	/* apple_hyb */
 	}
-	closedir(current_dir);
+	free(d_list);
 
 #ifdef apple_hyb
 	/*


